security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -477,7 +477,12 @@ func (sc *SecretController) generateKeyAndCert(saName string, saNamespace string
 	}
 
 	certChainPEM := sc.ca.GetCAKeyCertBundle().GetCertChainPem()
-	certPEM, signErr := sc.ca.Sign(csrPEM, strings.Split(id, ","), sc.certTTL, sc.forCA)
+
+	// Check if time duration until (intermediate) certificate will expire is smaller than the workload TTL
```

Comment:
as i mentioned, if you redesign the function interface, this will be just implementation details, for accuracy. callers don't have to comment on the behaivor of the library.

Reply:
and in line 518, it's using GetWaitTime's returning error to see if the cert needs to be refreshed. If we don't check over there, this refreshSecret() -> generateKeyCert() call chains won't even be called, right?